winml/lib/Api.Image/EventTimer.h

```diff
@@ -0,0 +1,29 @@
+#include "pch.h"
+
+class EventTimer
+{
+public:
+  EventTimer()
+  {
+  }
+  
+  bool Start()
+  {
+    auto now = std::chrono::high_resolution_clock::now();
+    if (!_started || 
+         std::chrono::duration_cast<std::chrono::microseconds>(now - _startTime).count() > _kDurationBetweenSendingEvents)
+    {
+      _started = true;
+      _startTime = std::chrono::high_resolution_clock::now();
+      return true;
+    }
+
+    return false;
+  }
+
+private:
+  bool _started = false;
```

Comment:
`_started` isn't needed (`_startTime` is initialized as zero)

Reply:
well... I debated that for a bit with myself. I agree it's unlikely that the user will hit this code within 50 microseconds of booting the system, but I also didn't want to take a dependency on the implementation detail. Also, the documentation makes it clear that the class is to measure durations and not absolutes, so I didn't want to count on an absolute value. 

But I'm not super committed to this line of reasoning :). what do you think?

Reply:
I usually think less code is clearer and more expressive, but I'm fine with the current state